test/k8s/net_policies.go

```diff
@@ -370,6 +378,147 @@ var _ = SkipDescribeIf(func() bool {
+			Expect(ok).Should(BeTrue(), "Cannot get app1 pod IP")
+
+			_, err = kubectl.CiliumPolicyAction(
+				namespaceForTest, ICMPv6Policy, helpers.KubectlApply, helpers.HelperTimeout)
+			Expect(err).Should(BeNil())
+
+			trace = kubectl.CiliumExecMustSucceed(context.TODO(), ciliumPod, fmt.Sprintf(
+				"cilium policy trace --src-k8s-pod %s:%s --dst-k8s-pod %s:%s --dport 128/ICMPv6",
+				namespaceForTest, appPods[helpers.App2], namespaceForTest, appPods[helpers.App1]))
+			trace.ExpectContains("Final verdict: ALLOWED", "Policy trace output mismatch")
+
+			trace = kubectl.CiliumExecMustSucceed(context.TODO(), ciliumPod, fmt.Sprintf(
+				"cilium policy trace --src-k8s-pod %s:%s --dst-k8s-pod %s:%s --dport 128/ICMPv6",
+				namespaceForTest, appPods[helpers.App3], namespaceForTest, appPods[helpers.App1]))
+			trace.ExpectContains("Final verdict: DENIED", "Policy trace output mismatch")
+
+			res = kubectl.ExecPodCmd(
+				namespaceForTest, appPods[helpers.App2],
+				helpers.Ping6(app1IP))
+			res.ExpectSuccess("%q cannot ping to %q", appPods[helpers.App2], appPods[helpers.App1])
+
+			res = kubectl.ExecPodCmd(
+				namespaceForTest, appPods[helpers.App3],
+				helpers.Ping6(appPods[helpers.App1]))
+			res.ExpectFail("%q can ping to %q", appPods[helpers.App3], appPods[helpers.App1])
+
+			By("Testing ICMPv6 deny rules")
+
+			_, err = kubectl.CiliumPolicyAction(
+				namespaceForTest, ICMPv6PolicyDeny, helpers.KubectlApply, helpers.HelperTimeout)
+			Expect(err).Should(BeNil())
+
+			trace = kubectl.CiliumExecMustSucceed(context.TODO(), ciliumPod, fmt.Sprintf(
+				"cilium policy trace --src-k8s-pod %s:%s --dst-k8s-pod %s:%s --dport 128/ICMPv6",
+				namespaceForTest, appPods[helpers.App2], namespaceForTest, appPods[helpers.App1]))
+			trace.ExpectContains("Final verdict: DENIED", "Policy trace output mismatch")
+
+			trace = kubectl.CiliumExecMustSucceed(context.TODO(), ciliumPod, fmt.Sprintf(
+				"cilium policy trace --src-k8s-pod %s:%s --dst-k8s-pod %s:%s --dport 128/ICMPv6",
+				namespaceForTest, appPods[helpers.App3], namespaceForTest, appPods[helpers.App1]))
+			trace.ExpectContains("Final verdict: DENIED", "Policy trace output mismatch")
+
+			res = kubectl.ExecPodCmd(
+				namespaceForTest, appPods[helpers.App2],
+				helpers.Ping6(app1IP))
+			res.ExpectFail("%q can ping to %q", appPods[helpers.App2], appPods[helpers.App1])
+
+			res = kubectl.ExecPodCmd(
+				namespaceForTest, appPods[helpers.App3],
+				helpers.Ping6(appPods[helpers.App1]))
+			res.ExpectFail("%q can ping to %q", appPods[helpers.App3], appPods[helpers.App1])
+
+			_, err = kubectl.CiliumPolicyAction(
+				namespaceForTest, ICMPv6Policy,
+				helpers.KubectlDelete, helpers.HelperTimeout)
+			Expect(err).Should(BeNil(), "Cannot delete ICMPv6 Policy")
+
+			_, err = kubectl.CiliumPolicyAction(
+				namespaceForTest, ICMPv6PolicyDeny,
+				helpers.KubectlDelete, helpers.HelperTimeout)
+			Expect(err).Should(BeNil(), "Cannot delete ICMPv6 Policy Deny")
```

Comment:
If the test fails on one of the previous commands, the ICMP policies won't be deleted, right? That is likely to cause issues in subsequent tests, so we should delete them in an AfterAll step or similar.

Reply:
Ah, nice! Yes, nothing else needed.